darwin/torch/dataset.py

```diff
@@ -343,3 +325,86 @@ def measure_weights(self, **kwargs):
             target = self.get_target(i)
             labels.extend([a["category_id"] for a in target["annotations"]])
         return self._compute_weights(labels)
+
+
+class BoundingBoxDetectionDataset(LocalDataset):
```

Comment:
I would prefer we have this class in a separate file and accompany it with its own tests. 
The `dataset.py` file is becoming increasingly larger. Dividing into smaller chunks would make it more manageable IMO.

Also tests. We all agreed each new PR should add a test, in this case, the new class does not have tests yet :D

Reply:
I agree different datasets should go in different files, this should be fixed by a different PR. Keep in mind testing dataset, in this case, is not trivial with the current darwin APIs.

I lost the test files, if you remember I had problems with my previous PR because `config.json` was still tracked. I don't think there is a way to get my files back and no time to write the tests from scratch again now. To be fair, nothing is tested there.